cli/lsp/language_server.rs

```diff
@@ -1023,6 +1067,41 @@ impl Inner {
       self.lint_options = lint_options;
       self.fmt_options = fmt_options;
       self.recreate_http_client_and_dependents().await?;
+      if let Some(config_file) = self.config.maybe_config_file() {
+        if let Ok((compiler_options, _)) = config_file.to_compiler_options() {
+          if let Some(compiler_options_obj) = compiler_options.as_object() {
+            if let Some(jsx_import_source) =
```

Comment:
Nitpick, but all of this should be simplified using `let else`

Reply:
Probably it could just be extracted out to a separate method?

Reply:
In this case IMO it's more readable like this, I don't want to add more methods for this because it's going to be moved for #21029.